frontend/src/components/sidebar/TramSidebarComponent.vue

```diff
@@ -1,16 +1,123 @@
 <script setup lang="ts">
 import SidebarComponent from "@components/sidebar/SidebarComponent.vue"
+import useTimeUtils from "@composables/useTimeUtils"
+import { simulation } from "@wails/go/models"
+import { GetTramDetails } from "@wails/go/simulation/Simulation"
+import { ref, watch } from "vue"
 
 const model = defineModel<boolean>({ required: true })
+
+const props = defineProps<{
+  tramID?: number
```

Comment:
Let's change this to `tramId` so it doesn't look weird when passing props (`tram-id` instead of `tram-i-d`)
```suggestion
  tramId?: number
```